### DISCUSSION TITLE
Pretty print floats

### DISCUSSION THREAD

**Discussion by 0xbentang** [tokio-rs/tracing#3088] [Q&A] [answered]
In the span I'll like to add a field to show some elapsed time, for example:

```
info!(elapsed = now.elapsed().as_secs_f64(), "Finished");
```

However it'll print a float like `0.088599778` which is not particularly human friendly. 

Right now I can do

```
info!(elapsed = format!("{:.4}s", now.elapsed().as_secs_f64()), "Finished");
```

To limit the no. of decimal places, as well as a suffix `s` to indicate it is in seconds. But `format!` will allocate a String every time and it'll not be efficient if I have a lot of logs? Is there any other way to print this more efficiently?

**Comment by mladedav** [comment, marked as answer]
It's not the best thing ever, but you can create a struct `PrettyFloat(f64)` and implement `std::fmt::Debug` on it so that it uses exactly the format you want.

Then you should be able to `info!(x  = debug(PrettyFloat(y)));` and it'll use your `Debug` implementation.

You could also do the same thing with `Display` if you'd prefer.

**Reply by 0xbentang** [reply]
Thank you this works nicely!
